src/GoodsList.tsx

```diff
@@ -1,16 +1,16 @@
-import React from 'react';
+import React, { memo } from 'react';
 import { Good } from './types/Good';
 
 type Props = {
-  goods: Good[]
+  goods: Good[];
 };
 
-export const GoodsList: React.FC<Props> = ({ goods }) => (
+export const GoodsList: React.FC<Props> = memo(({ goods }) => (
```

Comment:
HOC memo redundant here, because goods changes on each handleGet5First , handleGetAll , handleGetRedGoods functions call. More correctly wrap this functions call into useMemo